### PR TITLE
8297851: Add devkit for RISC-V

### PR DESCRIPTION
**Pull request by magicus**
We should add devkit support for linux/riscv64.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8297851](https://bugs.openjdk.org/browse/JDK-8297851): Add devkit for RISC-V

### Reviewers
 * [Fei Yang](https://openjdk.org/census#fyang) (@RealFYang - **Reviewer**) ⚠️ Review applies to [984358b3](https://git.openjdk.org/jdk/pull/11432/files/984358b31972e4ed697ce2eb567161bf740aef84)
 * [Erik Joelsson](https://openjdk.org/census#erikj) (@erikj79 - **Reviewer**) ⚠️ Review applies to [984358b3](https://git.openjdk.org/jdk/pull/11432/files/984358b31972e4ed697ce2eb567161bf740aef84)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/11432/head:pull/11432` \
`$ git checkout pull/11432`

Update a local copy of the PR: \
`$ git checkout pull/11432` \
`$ git pull https://git.openjdk.org/jdk pull/11432/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 11432`

View PR using the GUI difftool: \
`$ git pr show -t 11432`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/11432.diff">https://git.openjdk.org/jdk/pull/11432.diff</a>

</details>

### CODE REVIEW

File: make/devkit/Tools.gmk [hunk 1]
```diff
@@ -63,20 +63,28 @@ ifeq ($(BASE_OS), OL)
...
-  endif
-  ARCHIVED := $(shell [ $(BASE_OS_VERSION) -lt $(DEFAULT_OS_VERSION) ] && echo true)
-  ifeq ($(ARCHIVED),true)
-    BASE_URL := https://archives.fedoraproject.org/pub/archive/$(FEDORA_TYPE)/releases/$(BASE_OS_VERSION)/Everything/$(ARCH)/os/Packages/
+  ifeq ($(ARCH), riscv64)
+    DEFAULT_OS_VERSION := rawhide/68692
+    ifeq ($(BASE_OS_VERSION), )
+      BASE_OS_VERSION := $(DEFAULT_OS_VERSION)
+    endif
+    BASE_URL := http://fedora.riscv.rocks/repos-dist/$(BASE_OS_VERSION)/$(ARCH)/Packages/
```
Comment (RealFYang) [inline make/devkit/Tools.gmk:71]: Hi,  thanks for adding devkit support for linux/riscv64.
I tried and it seems that I could not access this URL: http://fedora.riscv.rocks/repos-dist/rawhide/68692/riscv64/Packages/
Anything I missed here?

**Comment by magicus** [comment]
Bot, gimme some slack.

File: make/devkit/Tools.gmk (hunk 1 above)
Comment (RealFYang) [reply make/devkit/Tools.gmk:71]: Looks like the URL are accessible now again. Changes looks good to me.

**Review by RealFYang — APPROVED** [review]

File: make/devkit/Tools.gmk (hunk 1 above)
Comment (magicus) [reply make/devkit/Tools.gmk:71]: The varying availablility is a bit unnerving, as is the fact that only a single build of rawhide is present. :-(

Clearly, this is not an ideal way to get an environment to build riscv in. In fact, the way GHA does it by using Debian's debootstrap is probably much better. The only reason for doing it this way is that it fits the current devkit model and thus do not require a complete rewrite of the devkit make files.

**Comment by magicus** [comment]
It turned out we have bugs when cross-compiling riscv, so my attempts at showing that this works in the Oracle CI system has been unsuccessful so far. But now I have a local fix for [JDK-8300550 ](https://bugs.openjdk.org/browse/JDK-8300550), and it seems promising.

**Comment by magicus** [comment]
The solution worked fine, and will be fixed with [JDK-8300550](https://bugs.openjdk.org/browse/JDK-8300550).

/integrate

**Pull request closed without merging**